include/sys/zil.h

```diff
@@ -393,6 +393,7 @@ typedef void (*zil_callback_t)(void *data);
 typedef struct itx {
 	list_node_t	itx_node;	/* linkage on zl_itx_list */
 	void		*itx_private;	/* type-specific opaque data */
+	uint64_t	itx_private2;
```

Comment:
```suggestion
	uint64_t	itx_gen;
```

Perhaps it would make more sense to add this a common member and rename it `itx_gen`?  We already do something similar in `zfs_log_rename` and `zfs_log_remove` where we set an `itx->itx_oid`.  Then move it down after `itx_oid`.

Reply:
Right, I use private2 because it's used for callback argument just like private.
But I guess itx_gen makes more sense. I'll change that.